### PR TITLE
append primary signature to batch

### PR DESCRIPTION
**Pull request by HagarMeir**
https://github.com/hyperledger/fabric-x-orderer/issues/37

### CODE REVIEW

File: node/ledger/batch_encoding.go [hunk 1]
```diff
@@ -66,10 +66,13 @@ func (b *FabricBatch) ConfigSequence() types.ConfigSequence {
...
 }
 
-// PrimarySignature returns nil for now.
-// TODO: Store and retrieve the primary signature from block metadata.
+// PrimarySignature returns the primary signature from block metadata at SIGNATURES index.
 func (b *FabricBatch) PrimarySignature() []byte {
-	return nil
+	m := (*common.Block)(b).GetMetadata().GetMetadata()
+	if len(m) <= int(common.BlockMetadataIndex_SIGNATURES) {
+		return nil
```
[the remark this answers is not in the document]
Comment (tock-ibm) [reply node/ledger/batch_encoding.go:73]: This is as designed - batch encoding is incompatible with Fabric blocks.
Ignore.

File: node/ledger/batch_encoding.go [hunk 2]
```diff
@@ -112,7 +116,7 @@ func NewFabricBatchFromRequests(
 			Data: batchedRequests,
 		},
 		Metadata: &common.BlockMetadata{
-			Metadata: [][]byte{{}, {}, {}, buff, {}},
+			Metadata: [][]byte{primarySignature, {}, {}, buff, {}},
 		},
```
[the remark this answers is not in the document]
Comment (tock-ibm) [reply node/ledger/batch_encoding.go:120]: Ignore. Works as designed.

File: node/ledger/batch_ledger_array_test.go [hunk 3]
```diff
@@ -143,7 +143,7 @@ func TestBatchLedgerArrayPart(t *testing.T) {
 	for _, pID := range parties {
 		part := a.Part(pID)
 		for seq := uint64(0); seq < 10; seq++ {
-			part.Append(types.BatchSequence(seq), 0, batchedRequests)
+			part.Append(types.BatchSequence(seq), 0, batchedRequests, nil)
```
Comment (tock-ibm) [inline node/ledger/batch_ledger_array_test.go:146]: add a test case with non-nil sig and verify retrieval here as well, not just in the `_part_test.go`

**Review by tock-ibm — APPROVED** [review]
